odoo/addons/base/controllers/rpc.py

```diff
@@ -47,7 +47,7 @@ def _xmlrpc(self, service):
         data = request.httprequest.get_data()
         params, method = loads(data)
         result = dispatch_rpc(service, method, params)
-        return dumps((result,), methodresponse=1, allow_none=False)
+        return dumps((result,), methodresponse=1, allow_none=True)
```

Comment:
I'm not particularly happy about changing the hardcoding here, but context detection is done inside of the check decorator in odoo/service/model.py. This line is specific to the XML-RPC implementation, so it could be removed and a working JSON-RPC solution would remain.